backend/src/main/java/com/redhat/cloud/notifications/routers/sources/SecretUtils.java

```diff
@@ -0,0 +1,224 @@
+package com.redhat.cloud.notifications.routers.sources;
+
+import com.redhat.cloud.notifications.models.BasicAuthentication;
+import com.redhat.cloud.notifications.models.Endpoint;
+import com.redhat.cloud.notifications.models.EndpointProperties;
+import com.redhat.cloud.notifications.models.SourcesSecretable;
+import io.quarkus.logging.Log;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@ApplicationScoped
+public class SecretUtils {
+
+    /**
+     * Used to manage the secrets on Sources.
+     */
+    @Inject
+    @RestClient
+    SourcesService sourcesService;
+
+    /**
+     * Loads the endpoint's secrets from Sources.
+     * @param endpoint the endpoint to get the secrets from.
+     */
+    public void loadSecretsForEndpoint(Endpoint endpoint) {
+        EndpointProperties endpointProperties = endpoint.getProperties();
+
+        if (endpointProperties instanceof SourcesSecretable) {
+            var props = (SourcesSecretable) endpointProperties;
+
+            final Long basicAuthSourcesId = props.getBasicAuthenticationSourcesId();
+            if (basicAuthSourcesId != null && basicAuthSourcesId > 0) {
+                final Secret secret = this.sourcesService.getById(basicAuthSourcesId);
+
+                props.setBasicAuthentication(
+                    new BasicAuthentication(
+                        secret.username,
+                        secret.password
+                    )
+                );
+            }
+
+            final Long secretTokenSourcesId = props.getSecretTokenSourcesId();
+            if (secretTokenSourcesId != null && secretTokenSourcesId > 0) {
+                final Secret secret = this.sourcesService.getById(secretTokenSourcesId);
+
+                props.setSecretToken(secret.password);
+            }
+        }
+    }
+
+    /**
+     * Creates the endpoint's secrets in Sources.
+     * @param endpoint the endpoint to create the secrets from.
+     */
+    public void createSecretsForEndpoint(Endpoint endpoint) {
+        EndpointProperties endpointProperties = endpoint.getProperties();
+
+        if (endpointProperties instanceof SourcesSecretable) {
+            var props = (SourcesSecretable) endpointProperties;
+
+            final BasicAuthentication basicAuth = props.getBasicAuthentication();
+            if (basicAuth != null) {
```

Comment:
`basicAuth` could contains null/empty fields. We should probably not create anything in Sources if there's no actual data to put there.

Reply:
Solved!